full-moon/src/ast/visitors.rs

```diff
@@ -431,3 +431,137 @@ impl<'a> VisitMut<'a> for LocalAssignment<'a> {
         self
     }
 }
+
+impl<'a> Visit<'a> for GenericFor<'a> {
+    fn visit<V: Visitor<'a>>(&self, visitor: &mut V) {
+        visitor.visit_generic_for(self);
+        self.for_token.visit(visitor);
+
+        let mut type_specifiers;
```

Comment:
```suggestion
        let type_specifiers;
```

Reply:
This needs to be labelled as `mut` I think because `next()` is being called on the iterator later, mutating it